### PR TITLE
Short redirection syntax

### PR DESCRIPTION
**Pull request by dandavison**
This one's pretty clear from the diff! It's more of a language design discussion.

### CODE REVIEW

**Comment by fdncred** [comment]
IMO, very ergonomic UX tweak for just 3 lines of code. +1

File: crates/nu-parser/src/lex.rs [hunk 1]
```diff
@@ -247,21 +247,21 @@ pub fn lex_item(
...
-        b"err>" => (
+        b"err>" | b"e>" => (
             Token {
                 contents: TokenContents::ErrGreaterThan,
                 span,
             },
             None,
         ),
-        b"out+err>" | b"err+out>" => (
+        b"out+err>" | b"err+out>" | b"o+e>" | b"e+o>" => (
```
Comment (dandavison) [inline crates/nu-parser/src/lex.rs:264]: Are there competing desires to parse spaceless math here @jntrnr? If the `+` is impacting math parsing we could use `oe>` and `eo>`.

**Comment by amtoine** [comment]
maybe `out+e>`, `e+out>`, `o+err>` and `err+o>` should be valid as well for the third case? :open_mouth:

**Comment by dandavison** [comment]
> maybe `out+e>`, `e+out>`, `o+err>` and `err+o>` should be valid as well for the third case? 😮

Hm, what would be the argument for that?

**Comment by amtoine** [comment]
> Hm, what would be the argument for that?
>
not much, just some thoughts :yum: 

- maybe for completeness?
- if i write a command with `out>` and i want to add `err` in a compact way, maybe allowing `out+e>` would save some editing?

to be honest i do not really use this feature :wink:

**Comment by dandavison** [comment]
> not much, just some thoughts 

OK, no, I think it's sufficient for users to use long or short syntax but not a mixture.

> to be honest i do not really use this feature

How do you write output to file? Do you prefer `| save` to `out>`?

**Comment by dandavison** [comment]
> not much, just some thoughts 

OK, no, I think it's sufficient for users to use long or short syntax but not a mixture.

> to be honest i do not really use this feature

How do you write output to file? Do you prefer `| save` to `out>`?

**Comment by amtoine** [comment]
> OK, no, I think it's sufficient for users to use long or short syntax but not a mixture.
> 
that's sensible :+1: 

> How do you write output to file? Do you prefer `| save` to `out>`?
>
yes, always `| save` :yum:

File: crates/nu-parser/src/lex.rs (hunk 1 above)
Comment (jntrnr) [reply crates/nu-parser/src/lex.rs:264]: We talked about it, but I think it'd get tricky with `+` and `-` being allowed in filenames and `*` being a valid glob character. I'm guessing we'll keep with requiring spaces.

**Comment by jntrnr** [comment]
👍 from me

**Pull request merged**